### ISSUE TITLE
Chrome Browser Profile / Cookies not applying to SingleFile in v0.7.2?

### ISSUE THREAD

**Issue by JitteryDoodle** [ArchiveBox/ArchiveBox#1445]
Hello,

I've been able to get cookies working for the Chrome > PDF, Screenshot, Headers, Chrome > HTML, and Readability views, but SingleFile is not signed in. This applies to multiple websites. What could the issue be and how can I troubleshoot this? I see nothing incorrect in the logs, even the logs related to singlepage display my user_data_dir path.

Here's my version info (confusingly on part, I named the chromium mount "chrome_profile", but this is actually the full chromium folder with the Default folder inside):
```
# archivebox --version
0.7.2
ArchiveBox v0.7.2 COMMIT_HASH=315c9f3 BUILD_TIME=2024-04-24 22:47:02 1713998822
IN_DOCKER=True IN_QEMU=False ARCH=x86_64 OS=Linux PLATFORM=Linux-6.1.74-Unraid-x86_64-with-glibc2.36 PYTHON=Cpython
FS_ATOMIC=True FS_REMOTE=True FS_USER=0:0 FS_PERMS=644
DEBUG=False IS_TTY=True TZ=UTC SEARCH_BACKEND=ripgrep LDAP=False

[i] Dependency versions:
 √  PYTHON_BINARY         v3.11.9         valid     /usr/local/bin/python3.11                                                   
 √  SQLITE_BINARY         v2.6.0          valid     /usr/local/lib/python3.11/sqlite3/dbapi2.py                                 
 √  DJANGO_BINARY         v3.1.14         valid     /usr/local/lib/python3.11/site-packages/django/__init__.py                  
 √  ARCHIVEBOX_BINARY     v0.7.2          valid     /usr/local/bin/archivebox                                                   

 √  CURL_BINARY           v8.5.0          valid     /usr/bin/curl                                                               
 √  WGET_BINARY           v1.21.3         valid     /usr/bin/wget                                                               
 √  NODE_BINARY           v20.12.2        valid     /usr/bin/node                                                               
 √  SINGLEFILE_BINARY     v1.1.46         valid     /app/node_modules/single-file-cli/single-file                               
 √  READABILITY_BINARY    v0.0.11         valid     /app/node_modules/readability-extractor/readability-extractor               
 √  MERCURY_BINARY        v1.0.0          valid     /app/node_modules/@postlight/parser/cli.js                                  
 √  GIT_BINARY            v2.39.2         valid     /usr/bin/git                                                                
 √  YOUTUBEDL_BINARY      v2023.12.30     valid     /usr/local/bin/yt-dlp                                                       
 √  CHROME_BINARY         v124.0.6367.29  valid     /usr/bin/chromium-browser                                                   
 √  RIPGREP_BINARY        v13.0.0         valid     /usr/bin/rg                                                                 

[i] Source-code locations:
 √  PACKAGE_DIR           24 files        valid     /app/archivebox                                                             
 √  TEMPLATES_DIR         3 files         valid     /app/archivebox/templates                                                   
 -  CUSTOM_TEMPLATES_DIR  -               disabled  None                                                                        

[i] Secrets locations:
 √  CHROME_USER_DATA_DIR  44 files        valid     /var/chrome_profile                                                         
 √  COOKIES_FILE          10.2 KB         valid     /var/chrome_profile/cookies.txt                                             

[i] Data locations:
 √  OUTPUT_DIR            7 files @       valid     /data                                                                       
 √  SOURCES_DIR           157 files       valid     ./sources                                                                   
 √  LOGS_DIR              1 files         valid     ./logs                                                                      
 √  ARCHIVE_DIR           293 files       valid     ./archive                                                                   
 √  CONFIG_FILE           81.0 Bytes      valid     ./ArchiveBox.conf                                                           
 √  SQL_INDEX             4.0 MB          valid     ./index.sqlite3
 ```

Willing to share more if needed.

Thanks!

**Comment by pirate** [comment]
I have gotten it working on 0.7.x so I know it's possible, but haven't tested it lately (chrome updates often and sometimes breaks things)!

I'm super deep working on the v0.8 release that's coming up right now, but I'll try to make sure cookies for singlefile are working smoothly in the next version if they aren't currently.

**Comment by pirate** [comment]
So I just double checked, in 0.7.x chrome profile use with singlefile should work automatically... the code is there and it's working for me on some sites. Unfortunately many larger (big tech) sites detect headless browsers to block bots and log you out automatically. Can you share some of the domains where you're getting blocked and I can try to investigate further?

You can also try using cookies.txt and tweaking your `CHROME_USER_AGENT` to see if that helps:

To use cookies with singlefile on v0.7.x you'd do:
```bash
# to use cookies.txt:
archivebox config --set SINGLEFILE_ARGS=["--browser-cookies-file=/data/cookies.txt"]
```

On [v0.8.0+](https://github.com/ArchiveBox/ArchiveBox/releases/tag/v0.8.0-rc) (coming soon) it'll work automatically without needing that ^ if you have [`COOKIES_FILE`](https://github.com/ArchiveBox/ArchiveBox/wiki/Configuration#cookies_file) setup.


More info:
- https://github.com/ArchiveBox/ArchiveBox/pull/1372
- https://github.com/ArchiveBox/ArchiveBox/pull/1021

**Comment by iluvatyr** [comment]
I tested with my docker archivebox and created the chrome profile within docker using the novnc container. Whenever I start the chromium-browser and check the login in novnc, it is actually working and Im logged in. Archiving a page, it is as if there was no chrome profile...
Environment variables are all set.

**Comment by iluvatyr** [comment]
so any fix upcoming?

**Comment by abrasumente233** [comment]
As far as I know this is not an ArchiveBox issue. IIRC chromium simply won't load profiles when launching headless. I can reproduce the issue on my Mac, launching chromium while datadir, with or without headless. I don't know the motivation and workaround though.

**Comment by abrasumente233** [comment]
This doesn't seem to be an ArchiveBox problem. From what I remember, Chromium just won't load profiles when it's launched headless. I can replicate this on my Mac, launching Chromium with a data directory, headless on/off. Not sure why it's designed this way or how to work around it.

**Comment by mstarodub** [comment]
I'm experiencing the polar opposite of this (at least with the 0.8rc). Singlefile seems to be logged in, but everything else isn't. 

I've verified that the profile is set up correctly by running with `CHROME_HEADLESS=False` - opened chromium with `--user-data-dir=...` prior to running `archivebox add`, and it opened the archived site in the same session.

However, when running archivebox with the chromium instance closed, it gets logged out. Even weirder is that a systemwide extension I manually disabled in the dedicated archivebox profile gets re-enabled?! So I have reason to believe it somehow resets the profile completely.

I've been debugging this for hours and at this point the only thing I haven't done yet is reading through chromium / archivebox source code. Would really appreciate some help as the archival matter at hand is quite urgent

**Comment by lamons** [comment]
I am having the similar issue here, in novnc everything seems working fine, but every archive method failed (stuck in the login page). I tried having `CHROME_HEADLESS=False` with novnc on and can actually see the chrome window pop up with successfully logged in page, but the archive still showing a login page. I tried also running sing-file (with `--browser-headless=true`) inside the container and it  got the login page as well, if using `--browser-headless=false` the chrome windows poped up for a second and crashed for not sure what reason, but I can see in that second it's still the login page, so the profile probably failed to be loaded by single-file.

**Comment by TooManyStacks** [comment]
I am seeing the same, `image: archivebox/archivebox:latest' I tried to put a cookies file, that didn't work. Checked with cat in the container,the data was there.

Then tried to pass my chromium profile with the cookies accepted on the site I am testing on (ad.nl a newspaper). I now see it finally gets the page title but still the cookie pop up blocks the whole page. Tried adding a cookie accept extension, no dice.

Also tried some other sites,all of them claim I never accepted cookies.

**Comment by yosofbadr** [comment]
I am having the same issue, not sure if it is a misconfiguration on my part, if it is not supported, or if the website I am trying to archive is indeed logging me out.

Edit: Did some testing and it looks like the site detects chrome-driver and automatically logs out the account. A bit of a shame, are there any work arounds for this?

Edit2: I tested this in a small PoC, it was the site automatically logging me out, however I was able to get around this by injecting cookies after the site loaded then using SingleFile to download the page.

**Comment by TooManyStacks** [comment]
What I notice in on projects, like playwright. Is that cookies do not easily transfer from system to system and headless and with head act differently with cookies.
Maybe the same is true for some sites here as well?

**Comment by pirate** [comment]
Yes that is unfortunately true @TooManyStacks, many sites tie the session to a browser fingerprint and will invalidate it if they detect a differnt browser trying to re-use cookies. Adding other headers like the USER_AGENT can help but it's not perfect, that's why it's better to use a full `CHROME_USER_DATA_DIR` with exactly the same browser binary used to set it up (instead of only cookies), and always run in headful mode.

**Comment by TooManyStacks** [comment]
I tried with the chrome dir, but not headful mode. Will do that as a test later.

**Comment by pirate** [comment]
@claude can you fix this on dev, make sure singlefile uses the same SINGLEFILE_CHROME_ARGS (x-fallback: CHROME_ARGS) as the chrome plugin so that it gets the user data dir and other launch options.
